test/unit/controllers/Depmu_EntryController.test.js

```diff
@@ -0,0 +1,326 @@
+"use strict";
+var jhg = require('../../helpers/JsonHelperGenerator');
+var ValidationError = require('../../../api/lib/exceptions/ValidationError');
+var controller = require('../../../api/controllers/Depmu_EntryController');
+var moment = require('moment-timezone');
+
+var validTimestamp = moment().set({hour: 7, minute: 0, second: 0}).format();
+var pastTimestamp = moment(validTimestamp).subtract(1, 'millisecond').format();
+var futureTimestamp = moment(validTimestamp).add(1, 'day').format();
+
+describe('UNIT Depmu_EntryController', () => {
+  var res;
+
+  beforeEach(() => {
+    res = {
+      ok: sinon.stub()
+    };
+
+    sinon.stub(Prebooking, 'create').resolves({foo: 'bar'});
+    sinon.stub(Centres, 'getGenderAndCentreByCIDLocation').resolves({bar: 'raa'});
+    sinon.stub(Centres, 'update').resolves({bar: 'raa'});
+  });
+
+  afterEach(() => {
+    Prebooking.create.restore();
+    Centres.getGenderAndCentreByCIDLocation.restore();
+    Centres.update.restore();
+  });
+
+  describe('prebookingOptions', () => {
+    it('should pass the schema to res.ok', () => {
+      controller.prebookingOptions(null, res);
+      expect(res.ok).to.be.calledWith(DepmuEntryPrebookingValidatorService.schema);
+    });
+  });
+
+  describe('formatPrebooking', () => {
+    var dummyPrebooking = {
+      "cid_id": '134',
+      "location": ' EXAMPLE 1 ',
+      "task_force": ' EXAMPLE 2 '
+    };
+
+    it('should make the prebooking location lowercase and trimmed', () =>
+      expect(controller.formatPrebooking(dummyPrebooking)["location"]).to.eql('example 1')
+    );
+
+    it('should make the prebooking task force  lowercase and trimmed', () =>
+      expect(controller.formatPrebooking(dummyPrebooking)["task_force"]).to.eql('example 2')
+    );
+
+    it('should make the prebooking cid an integer', () =>
+      expect(controller.formatPrebooking(dummyPrebooking)["cid_id"]).to.eql(134)
+    );
+
+    it('should make the prebooking cid null if undefined', () => {
```

Comment:
The DepmuEntryPrebookingValidatorService would return a 400 Bad Request: pattern: "^\\d*$"